scripts/packages/VSCodeServer/src/VSCodeServer.jl

```diff
@@ -69,23 +69,25 @@ include("debugger.jl")
 function serve(args...; is_dev=false, crashreporting_pipename::Union{AbstractString,Nothing}=nothing)
     conn = connect(args...)
     conn_endpoint[] = JSONRPC.JSONRPCEndpoint(conn, conn)
+    start_eval_backend()
     run(conn_endpoint[])
 
     @async try
         msg_dispatcher = JSONRPC.MsgDispatcher()
 
         msg_dispatcher[repl_runcode_request_type] = repl_runcode_request
+        msg_dispatcher[repl_interrupt_notification_type] = repl_interrupt_request
         msg_dispatcher[repl_getvariables_request_type] = repl_getvariables_request
         msg_dispatcher[repl_getlazy_request_type] = repl_getlazy_request
         msg_dispatcher[repl_showingrid_notification_type] = repl_showingrid_notification
         msg_dispatcher[repl_loadedModules_request_type] = repl_loadedModules_request
         msg_dispatcher[repl_isModuleLoaded_request_type] = repl_isModuleLoaded_request
-        msg_dispatcher[repl_startdebugger_notification_type] = (conn, params)->repl_startdebugger_request(conn, params, crashreporting_pipename)
+        msg_dispatcher[repl_startdebugger_notification_type] = (conn, params) -> repl_startdebugger_request(conn, params, crashreporting_pipename)
 
         while true
             msg = JSONRPC.get_next_message(conn_endpoint[])
 
-            if is_dev
+            @async if is_dev
```

Comment:
This *shouldn't* cause any problems, but is required for us to process multiple requests/notifications at the same time.

Reply:
I have to admit this makes me quite nervous because if affects absolutely everything, i.e. now the whole REPL code here needs to be reentrant safe, which is certainly not something we thought about when we wrote it initially...

But I also don't have a fantastic other idea ;) We could go back to the original model, where the code execution message is a notification that returns right away once the code running has been started, and then the server sends a notification back when the code has finished running, that way we wouldn't have these long-running code execution requests. I think that would limit the whole reentrant story to code execution, right?